client/components/Slide/SlideEditor/index.jsx

```diff
@@ -194,51 +244,69 @@ export default class SlideEditor extends React.Component {
-                                        scenarioId={scenarioId}
-                                        value={value}
-                                        onChange={v =>
-                                            this.onComponentChange(index, v)
-                                        }
-                                    />
-                                );
-                                const componentMenu = (
-                                    <EditorMenu
-                                        type="component"
-                                        items={{
-                                            save: {
-                                                onClick: (...args) => {
-                                                    this.updatedState(...args);
-                                                }
-                                            },
-                                            delete: {
-                                                onConfirm: () => {
-                                                    this.onDeleteComponent(
-                                                        index
-                                                    );
+                            <Sortable onChange={onChangeComponentOrder}>
+                                {components.map((value, index) => {
+                                    const { type } = value;
+                                    const { Editor, Display } = Components[
+                                        type
+                                    ];
+                                    const edit = (
+                                        <Editor
+                                            slideIndex={index}
+                                            scenarioId={scenarioId}
+                                            value={value}
+                                            onChange={v =>
+                                                this.onComponentChange(index, v)
+                                            }
+                                        />
+                                    );
+                                    const componentMenu = (
+                                        <EditorMenu
+                                            type="component"
+                                            items={{
+                                                save: {
+                                                    onClick: (...args) => {
+                                                        this.updateState(
+                                                            ...args
+                                                        );
+                                                    }
+                                                },
+                                                delete: {
+                                                    onConfirm: () => {
```

Comment:
Can we do curryable here? Like we did in the previous PR?

Reply:
Done